### ISSUE TITLE
Open in OS' default browser

### ISSUE THREAD

**Issue by elmimmo** [adampresley/sublime-view-in-browser#22]
I can already choose what browser I can use by default in the operating system. While I see how it can be convenient for some people to choose a default for developing different from the OS default, I think it would be nevertheless better to just default to the system-wide default, instead of Firefox just because.

This is how I set my user preferences to "open in default browser, whichever it is at any time" in OS X, in case you want to incorporate it in the default preferences. I do not know the necessary command lines to do that in other OS.

```
{
    "supportedBrowsers": {
        "default": [
            {
                "osname": "posix",
                "platform": "darwin",
                "command": "open"
            }
        ]
    },

    "selectedBrowser": "default"
}
```

**Comment by TrevorSayre** [comment]
To open in your system default browser for OS X, the current `View In Browser > Settings - User` should contain:
```
{
  "posix": {
    "darwin": {
      "default": "open"
    }
  },
  "browser": "default"
}
```

**Comment by markljackson** [comment]
I'm trying ST again, not by choice, and I'm trying to make it work for me. It's once again frustrating, as I'm a TextMate user, and ST just seems to do things in an odd way.

Could someone explain why this won't work (to set Safari as default) in View in Browser — User?

`
[ 
       // tried with and without parenthesis 
	{"browser": "safari"} 
]
` 

In trevor's post he does the below, could someone explain why?

`
{ 
  "posix": { 
    "darwin": {
      "default": "open"
    }
  },
  "browser": "default"
}
`

**Comment by TrevorSayre** [comment]
To format code on GitHub: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code

macOS is [POSIX](https://en.wikipedia.org/wiki/POSIX) compliant and runs on [Darwin](https://en.wikipedia.org/wiki/Darwin_(operating_system)). In the snippet I provided, I set the option for `default` on a POSIX machine running Darwin to use the command `open`, the Darwin command for opening files on macOS. When you pass a URI to the `open` command, it opens the URI in your default browser.

The syntax can be seen in the [View In Browser.sublime-settings](https://github.com/adampresley/sublime-view-in-browser/blob/3297c91f629641794550f948e76f89b33824cbd9/View%20In%20Browser.sublime-settings) file.

So the snippet I provided simply defines a command for `default` which allows you to set `"browser": "default"`.

However, it also wipes out the command definitions for all the other browsers. So if you don't need an option for `default`, then your `Sublime Text > Preferences > Package Settings > View In Browser > Settings - User` should contain only:

```JSON
{
  "browser": "safari"
}
```

If you want support for a `default` as well as all the other browsers, then your `Sublime Text > Preferences > Package Settings > View In Browser > Settings - User` should contain:

```JSON
{
  "posix": {
    "darwin": {
      "default": "open",
      "firefox": "open -a \"/Applications/Firefox.app\"",
      "safari": "open -a \"/Applications/Safari.app\"",
      "chrome": "open -a \"/Applications/Google Chrome.app\""
    }
  },
  "browser": "default"
}
```

This will allow you to open with your default browser using `control+option+v`, but also open with a specific browser using `control+option+s` for Safari, `control+option+f` for Firefox, and `control+option+c` for Chrome.

**Comment by markljackson** [comment]
OK, I wasn't clear. Sorry. Let me try again.

Markdown; yea, I know Markdown. I just missed the single tick in the "insert code" selector. I was quite tired, when I wrote the post. Thanks for the link.

 Mac OS, POSIX, UNiX Darwin; yea, got that.

My question was directed toward why ST seemed to acting differently with "view in browser" than it does with other plugins. At least the few that I've used, and modified.

### What I was expecting to work

SUBLIME-KEYMAP has this format:

```xml

[
    {"keys": ["modifier + key"], "command": "do _this"}
]

```

Modifying that setting would would be this:

```xml

[
   // yes I know negative commands don't exist, illustration only
    {"keys": ["modifier + key"], "command": "don't_do_that"}  
]

```

I know that works, because I've done it. And that is what I was expecting to work in the USER file.

### What I tried that failed

I expected to put the following in the USER file …

1. Code snippet 1
```json

{
  "browser": "safari"
}

```

and override the ONE setting you built in (below)

2. Code Snippet 2
```json

{
   […]
	"browser": "firefox"
}

```

That didn't work (duh). It sounds like your saying that when I put the first code snippet in, it COMPLETELY wipes out the default settings. Is that right?

**Comment by TrevorSayre** [comment]
> That didn't work (duh). It sounds like your saying that when I put the first code snippet in, it COMPLETELY wipes out the default settings. Is that right?

Yep :) That is, if you put

```JSON
{
  "posix": {
    "darwin": {
    }
  }
}
```

That wipes out all the browser settings for macOS.

But if you just put:

```JSON
{
  "browser": "safari"
}
```

Then you should be all good. That will only override the `browser` key and not the macOS browser settings.

Also, this should work in your User key bindings configuration file:

```JSON
[
  { "keys": [ "ctrl+alt+v" ], "command": "view_in_browser", "args": { "browser": "safari" } }
]
```

But the default `control+option+s` should open Safari by default.

**Comment by markljackson** [comment]
Alright, thanks for the thorough response. I must have done something wrong when setting

```json

{
    "browser": "safari"
}

```

I'll try it again tonight.
